portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -46,11 +47,34 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         Query query = new Query(DATASTORE_LABEL);
         PreparedQuery results = datastore.prepare(query);
 
-        ArrayList<String> commentList = new ArrayList<String>();
-        for (Entity entity : results.asIterable()) {
-            commentList.add((String) entity.getProperty(INPUT_NAME));
+        int numComments;
+        // Get the number of comments the user wants to view. 
+        try {
+            numComments = Integer.parseInt(request.getParameter(NUM_COMMENTS_INPUT));
+        } catch (NumberFormatException e) {
+            numComments = 20;
         }
 
+        ArrayList<Map<String, String>> commentList = new ArrayList<Map<String, String>>();
```

Comment:
Looks like you're effectively using Map<String, String> as a data structure to store a comment.

I think it would be nicer to just make a simple comment class that contains an ID field and a comment field

Reply:
You could also use datastore.Entity for this

Reply:
Might change at the end of the week. I understand the reasoning behind it, but I would like to begin on the Libraries instead of changing things at the moment.